### DISCUSSION TITLE
Feature Request: Getting the previous Neighbour

### DISCUSSION THREAD

**Discussion by julianpollmann** [explosion/spaCy#10057] [Help: Other Questions] [answered]
The Token class has a [nbor(i) method](https://github.com/explosion/spaCy/blob/master/spacy/tokens/token.pyx#L175-L185) for retrieving one or more next token(s).
This method only allows to get next neighbours, therefore a feature request for a method to get the previous neighbours.

Token.lefts/rights do not cover this scenario.

**Comment by honnibal** [comment, marked as answer]
`token.nbor(-1)` should do what you're looking for.

**Reply by julianpollmann** [reply]
This raised an IndexError Exception, but should be expected. I had to check the Index against the parent, i.e. doc:

```
for token in doc:
    if token.i > 1:
        prev = token.nbor(-1)
        
    if token.i < len(doc) -1 :
        next = token.nbor()
```
